File: 14 linkedlist as class/Elem.java

public class Elem {
    private int value;
    private Elem next;

    public Elem(){}

    public Elem(int value, Elem next) {
        this.value = value;
        this.next = next;
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }

    public Elem getNext() {
        return next;
    }

    public void setNext(Elem next) {
        this.next = next;
    }
}

File: 12 linked list input and output/elem.cpp
//Примитивная переменная передается по значению

//Ссылка передается по значению
//Объект передается по ссылке

В С++

typedef struct {
	int value,
	elem * next;
} elem;

int main() {
	
	elem * p = new elem;
	//(*p).value = 20;
	p -> value = 20;

	//
	delete p;
	
	elem q;
	q.value = 20;

}